router/internal/eventorder/eventorder.go

```diff
@@ -67,35 +76,40 @@ func (b *Barrier) Enter(key string, jobID int64) (accepted bool, previousFailedJ
 	defer b.mu.RUnlock()
 	barrier, ok := b.barriers[key]
 	if !ok {
-		return true, nil // no barrier, accept
+		if b.concurrencyLimit == 0 { // if not concurrency limit is set accept the job
+			return true, nil
+		}
+		// create a new barrier with the concurrency limiter enabled
+		barrier = &barrierInfo{
+			concurrencyLimiter: make(map[int64]struct{}),
+		}
+		b.barriers[key] = barrier
```

Comment:
Is it OK to write into `b.barriers` even though on line 75 only a read lock was acquired?

Reply:
thanks, need to change the lock type to full